src/log/LogHandler.cpp

```diff
@@ -7,51 +7,55 @@
 
 static void handler(int sig)
 {
+    (void)sig;
     alarm(1);
     LOGFLUSH();
 }
 
-RAClog::FileLogHandler::FileLogHandler()
+namespace RAClog
 {
-    this->CreateFileLog();
-    sigset(SIGALRM, handler);
-    alarm(1);
-}
-
-RAClog::FileLogHandler::~FileLogHandler()
-{
-    this->CloseFileLog();
-}
 
-void RAClog::FileLogHandler::CreateFileLog()
-{
-    if (!oFileStream.is_open())
+    FileLogHandler::FileLogHandler()
     {
-	std::stringstream fileStream;
+	this->CreateFileLog();
+	sigset(SIGALRM, handler);
+	alarm(1);
+    }
 
-	fileStream << "log/" << RACtime::TimeHandler::GetTime() << "_" << pFileLogSuffix;
+    FileLogHandler::~FileLogHandler()
+    {
+	this->CloseFileLog();
+    }
 
-	oFileStream.open(fileStream.str());
-	try
-	{
-	    oFileStream.exceptions(oFileStream.failbit);
-	}
-	catch (const std::ios_base::failure& e)
+    void FileLogHandler::CreateFileLog()
+    {
+	if (!oFileStream.is_open())
 	{
-	    std::cerr << "An Error has occured during the creation of the file in the folder log" << std::endl <<  
-	    e.what() << std::endl;
-	    FileLogHandler::DeleteInstance();
+	    std::stringstream fileStream;
+
+	    fileStream << "log/" << RACtime::TimeHandler::GetTime() << "_" << pFileLogSuffix;
+	    oFileStream.open(fileStream.str());
+	    try
+	    {
+		oFileStream.exceptions(oFileStream.failbit);
+	    }
+	    catch (const std::ios_base::failure& e)
+	    {
+		std::cerr << "An Error has occured during the creation of the file in the folder log" << std::endl <<  
+		    e.what() << std::endl;
+		FileLogHandler::DeleteInstance();
+	    }
+
+	    LOG(INFO, "File Log Created");
```

Comment:
You go there event if you catch an error and delete the `FileLogHandler` instance. You should either rethrow or move this line in the try block

Reply:
yes